### DISCUSSION TITLE
有点不太懂knowhere,这个测试怎么显示GPU比CPU慢

### DISCUSSION THREAD

**Discussion by Escapethebug** [milvus-io/milvus#26779] [Q&A and General discussion]
Randomness seeded to: 3663026999
0.001 s: check invalid json keys
0.001 s: Test config json parse
0.000 s: check invalid json keys
0.000 s: Test config json parse
0.001 s: check invalid json keys
0.001 s: Test config json parse
0.000 s: check invalid json keys
0.000 s: Test config json parse
0.000 s: check invalid json keys
0.001 s: Test config json parse
0.000 s: check invalid json keys
0.000 s: Test config json parse
0.000 s: check invalid json keys
0.000 s: Test config json parse
0.000 s: check invalid json keys
0.000 s: Test config json parse
0.001 s: Check the json for the specific index
0.001 s: Test config json parse
0.000 s: check flat index config
0.000 s: Test config json parse
0.000 s: check ivf index config
0.000 s: Test config json parse
0.000 s: check hnsw index config
0.001 s: Test config json parse
0.001 s: check diskann index config
0.001 s: Test config json parse
2.913 s: Test Gpu Index Search
3.600 s: Test All GPU Index
3.360 s: Test Gpu Index Search
3.416 s: Test All GPU Index
0.368 s: Test Gpu Index Search
0.424 s: Test All GPU Index
0.263 s: Test Gpu Index Serialize/Deserialize
0.319 s: Test All GPU Index
1.089 s: Test Gpu Index Serialize/Deserialize
1.199 s: Test All GPU Index
3.893 s: Test Gpu Index Serialize/Deserialize
4.010 s: Test All GPU Index
0.979 s: Test Gpu Index Serialize/Deserialize
1.090 s: Test All GPU Index
0.000 s: Knowhere global config
0.000 s: Test distance_in_range
0.002 s: Test GetRangeSearchResult for HNSW/DiskANN
0.159 s: Test GetRangeSearchResult for Faiss
2.269 s: Test Cpu Index Search
2.269 s: Test All Mem Index Search
0.386 s: Test Cpu Index Search
0.386 s: Test All Mem Index Search
0.682 s: Test Cpu Index Search
0.682 s: Test All Mem Index Search
0.963 s: Test Cpu Index Search
0.964 s: Test All Mem Index Search
7.240 s: Test Cpu Index Search
7.240 s: Test All Mem Index Search
2.769 s: Test Cpu Index Search
2.769 s: Test All Mem Index Search
0.368 s: Test Cpu Index Range Search
0.368 s: Test All Mem Index Search
0.800 s: Test Cpu Index Range Search
0.800 s: Test All Mem Index Search
1.071 s: Test Cpu Index Range Search
1.074 s: Test All Mem Index Search
3.366 s: Test Cpu Index Range Search
3.367 s: Test All Mem Index Search
2.344 s: Test Cpu Index Serialize/Deserialize
2.344 s: Test All Mem Index Search
0.369 s: Test Cpu Index Serialize/Deserialize
0.369 s: Test All Mem Index Search
0.716 s: Test Cpu Index Serialize/Deserialize
0.716 s: Test All Mem Index Search
0.913 s: Test Cpu Index Serialize/Deserialize
0.913 s: Test All Mem Index Search
7.076 s: Test Cpu Index Serialize/Deserialize
7.076 s: Test All Mem Index Search
3.334 s: Test Cpu Index Serialize/Deserialize
3.334 s: Test All Mem Index Search
0.015 s: Test build IVFPQ with invalid params
0.015 s: Test All Mem Index Search
===============================================================================
All tests passed (176 assertions in 7 test cases)
![image](https://github.com/milvus-io/milvus/assets/45965328/c52c9859-587c-4788-aba7-b8faa9d30e2e)

**Comment by presburger** [comment]
@Escapethebug hi, which gpu card you use?

**Reply by Escapethebug** [reply]
Rtx 4090

**Reply by presburger** [reply]
ok. Compute Capability need add 89, https://github.com/zilliztech/knowhere/blob/66145d034f260cd7a277977ee533e8e26c76266a/CMakeLists.txt#L52

**Reply by Escapethebug** [reply]
Thanks you! I have modified it. What does this parameter mean? Is it related to this parameter?![image](https://github.com/milvus-io/milvus/assets/45965328/50eb6af0-ea01-4d89-adcf-c9485864f5da)
